SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx

```diff
@@ -1733,6 +1738,57 @@ void vtkSlicerSkeletalRepresentationInitializerLogic::DisplayResultSrep()
     AddModelNodeToScene(meshPoly, "skeletal mesh for initial object", true, 0, 0, 0);
     AddModelNodeToScene(curvePoly, "fold curve for initial object", true, 1, 1, 0);
 
+    // add spokeLength and spokeDirection to polyData
+    vtkSmartPointer<vtkPolyData> vtpUpSpoke = vtkSmartPointer<vtkPolyData>::New();
+    vtkSmartPointer<vtkPolyData> vtpDownSpoke = vtkSmartPointer<vtkPolyData>::New();
+    vtkSmartPointer<vtkPolyData> vtpCrestSpoke = vtkSmartPointer<vtkPolyData>::New();
+    CompletePolyData(upSpoke_poly, vtpUpSpoke);
+    CompletePolyData(downSpoke_poly, vtpDownSpoke);
+    CompletePolyData(crestSpoke_poly, vtpCrestSpoke, true);
+    // save to files
+    std::string outputUpFileName, outputDownFileName, outputCrestFileName;
+    outputUpFileName = mOutputPath + "/up.vtp";
+    outputDownFileName = mOutputPath + "/down.vtp";
+    outputCrestFileName = mOutputPath + "/crest.vtp";
+
+    vtkSmartPointer<vtkXMLPolyDataWriter> vtpWriter = vtkSmartPointer<vtkXMLPolyDataWriter>::New();
+    vtpWriter->SetDataModeToAscii();
+    vtpWriter->SetFileName(outputUpFileName.c_str());
+    vtpWriter->SetInputData(vtpUpSpoke);
+    vtpWriter->Update();
+
+    vtpWriter->SetFileName(outputDownFileName.c_str());
+    vtpWriter->SetInputData(vtpDownSpoke);
+    vtpWriter->Update();
+
+    vtpWriter->SetFileName(outputCrestFileName.c_str());
+    vtpWriter->SetInputData(vtpCrestSpoke);
+    vtpWriter->Update();
+
+    // compose header file
+    std::stringstream output;
+
+    output<<"<s-rep>"<<std::endl;
+    output<<"  <nRows>"<<ROWS<<"</nRows>"<<std::endl;
+    output<<"  <nCols>"<<COLS<<"</nCols>"<<std::endl;
+    output<<"  <meshType>Quad</meshType>"<< std::endl;
+    output<<"  <color>"<<std::endl;
+    output<<"    <red>0</red>"<<std::endl;
+    output<<"    <green>0.5</green>"<<std::endl;
+    output<<"    <blue>0</blue>"<<std::endl;
+    output<<"  </color>"<<std::endl;
+    output<<"  <isMean>False</isMean>"<<std::endl;
+    output<<"  <meanStatPath/>"<<std::endl;
+    output<<"  <upSpoke>"<< outputUpFileName<<"</upSpoke>"<<std::endl;
+    output<<"  <downSpoke>"<< outputDownFileName << "</downSpoke>"<<std::endl;
+    output<<"  <crestSpoke>"<< outputCrestFileName << "</crestSpoke>"<<std::endl;
+    output<<"</s-rep>"<<std::endl;
+
+    std::string header_file = mOutputPath + "/header.xml"; ;
+    std::ofstream out_file;
+    out_file.open(header_file);
+    out_file << output.rdbuf();
+    out_file.close();
```

Comment:
I am wondering if at some point a dedicated MRML node (and associated storage node), reader and writer should be added for s-rep.

Reply:
Is the format documented somewhere ?

Reply:
Sorry for the late reply. The format of the header file is documented in the user manual.

Reply:
I see. Then, I suggest to add a reference to the manual from the source code. 

May be here: https://github.com/KitwareMedical/SlicerSkeletalRepresentation/blob/5e396efc4abca6753ecadd5a2a71ba8ac519430e/SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx#L1625-L1628